lib/modules/stats/country_statistics_api.rb

```diff
@@ -0,0 +1,118 @@
+class Stats::CountryStatisticsApi
+  BASE_URL = 'https://dopa-services.jrc.ec.europa.eu/services/d6dopa30/'.freeze
+  ENDPOINTS = {
+    representative: {
+      endpoint: 'habitats_and_biotopes/get_ecoregion_all_inds',
+      field: 'prot_perc'
+    },
+    well_connected: {
+      endpoint: 'administrative_units/get_country_all_inds',
+      field: 'protconn'
+    },
+    importance: {
+      endpoint: 'administrative_units/get_country_all_inds',
+      field: 'kba_avg_prot_perc'
+    }
+  }.freeze
+  ERRORS = {
+    representative: """
+      Country level stats cannot be fetched for the representative endpoint.
+      Please invoke this function without an iso code.
+    """,
+    httparty: 'We are sorry but something went wrong while connecting to the API.',
+    data: 'We are sorry but something went wrong while processing the data from the API.',
+  }.freeze
+
+  ISO3_ATTRIBUTE = 'country_iso3'.freeze
+  NAME_ATTRIBUTE = 'country_name'.freeze
+
+  def self.import(iso3=nil)
+    endpoints = ENDPOINTS.slice(:well_connected, :importance)
+    # Get stats for each endpoint
+    # Representative stat is exlcuded because that is a global level stat
+    endpoints.each do |name, attributes|
+      url = endpoint_url(attributes[:endpoint])
+      # Connect to the API and fetch the data
+      data = fetch(url, iso3)
+
+      # Return if there's an error
+      return data if data.is_a?(Hash) && data.key?(:error)
+
+      # Update stat for each country
+      data.each do |stat|
+        iso3 = stat[ISO3_ATTRIBUTE]
+        country = Country.find_by_iso_3(iso3)
+        unless country
+          Rails.logger.info(not_found_error('country', iso3))
+          next
+        end
+
+        field = attributes[:field]
+        statistic = country.country_statistic
+        unless statistic
+          Rails.logger.info(not_found_error('statistic', iso3))
+          next
+        end
+
+        statistic.update_attributes("#{name}" => stat[field])
+      end
+    end
+  end
+
+  def self.get_stats(endpoint, iso3=nil)
+    if iso3.present? && endpoint.to_s == 'representative'
+      raise ArgumentError, ERRORS[:representative]
+    end
+    url = endpoint_url(ENDPOINTS[endpoint.to_sym][:endpoint])
+    data = fetch(url, iso3)
+
+    return data if data.key?(:error)
+
+    format_data(data, endpoint)
+  end
+
+  private
+
+  def self.format_data(data, endpoint)
+    field = ENDPOINTS[endpoint.to_sym][:field]
+    if endpoint.to_s == 'representative'
+      _sum = data.inject(0) do |sum, x|
+        sum + x[field]
+      end
+      # TODO Need to confirm if this is the correct calculation
```

Comment:
Do we need the same confirm for line 40 if the `app/models/country_statistic.rb` file?? If this is the case I would add a comment there as well, If not just ignore this comment.

Reply:
Good point, I will add a comment there as well so that we can double check with the relevant people. Thanks!